amgut/lib/data_access/test/test_ag_data_access.py

```diff
@@ -64,13 +62,127 @@ def test_getAnimalParticipants(self):
+                'barcode': '000016713',
+                'sample_date': datetime.date(2014, 5, 30),
+                'site_sampled': 'Stool'},
+               {'status': None,
+                'sample_time': datetime.time(19, 30),
+                'notes': 'REMOVED',
+                'barcode': '000016496',
+                'sample_date': datetime.date(2014, 4, 29),
+                'site_sampled': 'Stool'},
+               {'status': None,
+                'sample_time': datetime.time(19, 30),
+                'notes': 'REMOVED',
+                'barcode': '000016497',
+                'sample_date': datetime.date(2014, 4, 29),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(10, 20),
+                'notes': 'REMOVED',
+                'barcode': '000004213',
+                'sample_date': datetime.date(2013, 10, 16),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(9, 50),
+                'notes': 'REMOVED',
+                'barcode': '000004214',
+                'sample_date': datetime.date(2013, 10, 14),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(12, 0),
+                'notes': 'REMOVED',
+                'barcode': '000004215',
+                'sample_date': datetime.date(2013, 10, 13),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(9, 30),
+                'notes': 'REMOVED',
+                'barcode': '000004216',
+                'sample_date': datetime.date(2013, 10, 15),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(14, 25),
+                'notes': 'REMOVED',
+                'barcode': '000004218',
+                'sample_date': datetime.date(2013, 10, 12),
+                'site_sampled': 'Stool'},
+               {'status': 'Received',
+                'sample_time': datetime.time(10, 15),
+                'notes': 'REMOVED',
+                'barcode': '000004219',
+                'sample_date': datetime.date(2013, 10, 17),
+                'site_sampled': 'Stool'}]
+        self.assertItemsEqual(res, exp)
 
     def test_getEnvironmentalSamples(self):
-        raise NotImplementedError()
+        res = self.ag_data.getEnvironmentalSamples("d6b0f287-b9d9-40d4-82fd-a8fd3db6c476")
+        exp = [{'status': None, 'sample_time': datetime.time(21, 45),
+                'notes': 'REMOVED', 'barcode': '000028433',
+                'sample_date': datetime.date(2015, 6, 7), 'site_sampled': None}]
+        self.assertItemsEqual(res, exp)
 
     def test_getAvailableBarcodes(self):
-        raise NotImplementedError()
+        res = self.ag_data.getAvailableBarcodes("d8592c74-9694-2135-e040-8a80115d6401")
```

Comment:
Need test for when login_id not in DB, and over multiple kits/single kit as two tests.